yt_dlp/extractor/rtlnl.py

```diff
@@ -141,3 +141,176 @@ def _real_extract(self, url):
+        self._sort_formats(formats)
+        return {
+            'id': video_id,
+            'title': self._og_search_title(webpage),
+            'description': self._og_search_description(webpage),
+            'formats': formats,
+            'subtitles': subtitles,
+            'thumbnail': self._og_search_thumbnail(webpage),
+        }
+
+
+class RTLLuArticleIE(RTLLuBaseIE):
+    IE_NAME = 'rtl.lu:article'
+    _VALID_URL = r'https?://www\.rtl\.lu/(?:\w+)/(?:\w+)/a/(?P<id>\d+)\.html'
+    _TESTS = [{
+        # Audio-only
+        'url': 'https://www.rtl.lu/sport/news/a/1934360.html',
+        'info_dict': {
+            'id': '1934360',
+            'ext': 'mp3',
+            'thumbnail': 'https://static.rtl.lu/rtl2008.lu/nt/p/2022/06/28/19/e4b37d66ddf00bab4c45617b91a5bb9b.jpeg',
+            'description': 'md5:5eab4a2a911c1fff7efc1682a38f9ef7',
+            'title': 'md5:40aa85f135578fbd549d3c9370321f99',
+        }
+    }, {
+        # Video-only
+        'url': 'https://www.rtl.lu/kultur/news/a/1931683.html',
+        'info_dict': {
+            'id': '1931683',
+            'ext': 'mp4',
+            'description': 'md5:ad39b36e0039a109384b5996c373e835',
+            'title': 'Esch2022: Suessem ass déi nei "Gemeng vum Mount"',
+            'thumbnail': 'https://static.rtl.lu/rtl2008.lu/nt/p/2022/06/22/16/7f9d5141c40733ffd0054d1a4d01819e.jpeg',
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        # TODO: extract comment from https://www.rtl.lu/comments?status=1&order=desc&context=news|article|<video_id>
+        # we can context from <rtl-comments context=<context> in webpage
+        formats, subtitles = self.get_format(webpage, video_id)
+        self._sort_formats(formats)
+
+        return {
+            'id': video_id,
+            'title': self._og_search_title(webpage),
+            'description': self._og_search_description(webpage),
+            'formats': formats,
+            'subtitles': subtitles,
+            'thumbnail': self._og_search_thumbnail(webpage),
+        }
+
+
+class RTLLuTeleLiveIE(RTLLuBaseIE):
+    _VALID_URL = r'https?://www\.rtl\.lu/tele/live(?P<id>-\d+)?'
+    _TESTS = [{
+        'url': 'https://www.rtl.lu/tele/live',
+        'info_dict': {
+            'id': 'Tele:live',
+            'ext': 'mp4',
+            'live_status': 'is_live',
+            'title': r're:RTL - Télé LIVE \d{4}-\d{2}-\d{2} \d{2}:\d{2}',
+
+        }
+    }, {
+        'url': 'https://www.rtl.lu/tele/live-2',
+        'info_dict': {
+            'id': 'Tele:live-2',
+            'ext': 'mp4',
+            'live_status': 'is_live',
+            'title': r're:RTL - Télé LIVE \d{4}-\d{2}-\d{2} \d{2}:\d{2}',
+        }
+    }]
+
+    def _real_extract(self, url):
+        slug = '' if self._match_id(url) is None else self._match_id(url)
+        video_id = f'Tele:live{slug}'
```

Comment:
not a good idea to set id manually imo.
Just apply above suggestion and do video_id = self._match_id(url)